### PR TITLE
KVP basic test case - permissions and lsof checks additions

### PR DESCRIPTION
**Pull request by lixuemin2016**
…of check

Update existing ReadKVPData_BasicTest test case, locally we have manual test cases: 
1. check hypervkvpd pool file directory and file permission
#ls /var/lib/hyperv/.kvp_pool_*

2.  check number of opened file descriptors by hypervkvpd  #lsof \|grep -c kvp 
 check the number of process two minutes later #lsof \| grep -c kvp
Refer to bug https://bugzilla.redhat.com/show_bug.cgi?id=949664.

3. After RHEL7.3, ls -la /proc/`pidof hypervkvpd`/fd \| grep hv_kvp 

This case includes more test points, any comment, suggestions will update. Thank you so much.

### CODE REVIEW

File: WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh [hunk 1]
```diff
@@ -1,107 +1,141 @@
...
+        LogMsg "Error: Unable to detect OS architecture"
+        SetTestStateAborted
+        exit 60
+    fi
 fi
 
+chmod +x /root/kvp_client*
+poolCount=`/root/$kvp_client | grep -i pool | wc -l`
+if [ $poolCount -ne 5 ]; then
+	LogMsg "pools are not created properly"
```
Comment (chvalean) [inline WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh]: both LogMsg and UpdateSummary messages should match, I would suggest the following:
"Error: Could not find a total of 4 KVP pools!"

File: WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh [hunk 2]
```diff
@@ -1,107 +1,141 @@
...
 #
-if [ "${TC_COVERED:-UNDEFINED}" = "UNDEFINED" ]; then
-    msg="The test parameter TC_COVERED is not defined in ${CONSTANTS_FILE}"
-    echo $msg
-    echo $msg >> ~/summary.log
-    UpdateTestState $ICA_TESTABORTED
-    exit 30
+permCount=`stat -c %a /var/lib/hyperv/.kvp_pool* | grep 644 | wc -l`
+if [ $permCount -ne 5 ]; then
+        LogMsg ".kvp_pool file permission is incorrect "
```
Comment (chvalean) [inline WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh]: fix indentation

File: WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh [hunk 3]
```diff
@@ -1,107 +1,141 @@
...
-# This script will install and run the KVP client tool to verify that the KVP
-# pools are created and accessible.
-# Make sure we have kvptool.tar.gz file in Automation\..\lisa\Tools folder 
-# 
+# Description:
+# 1. verify that the KVP Daemon is running
+# 2. run the KVP client tool and verify that the data pools are created and accessible
+# 3. check kvp_pool file permission is 644
+# 4. check kernel version supports hv_kvp
+# 5. check lsof number for kvp after sleep 2 minutes
```
[the remark this answers is not in the document]
Comment (lixuemin2016) [reply WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh]: Thank you, have updated.

File: WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh [hunk 4]
```diff
@@ -1,107 +1,141 @@
...
+            ;;
+    esac
 
-LogMsg()
-{
-    echo `date "+%a %b %d %T %Y"` : ${1}    # To add the timestamp to the log file
+    if [ $? -ne 0 ]; then
+        LogMsg "ERROR: Failed to install lsof"
+        UpdateSummary "ERROR: Failed to install lsof"
+        SetTestStateAborted
```
[the remark this answers is not in the document]
Comment (lixuemin2016) [reply WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh]: Have updated by checking "command -v lsof", if already installed, "yum install -y lsof" will return 0. Here I'm not quite sure how to set the test result if lsof install fails and not fail the entire test case, so keep setTestStateAborted. Is it ok? Thank you.

File: WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh (hunk 1 above)
Comment (lixuemin2016) [reply WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh]: Changed based on comment, thank you.

File: WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh (hunk 2 above)
Comment (lixuemin2016) [reply WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh]: changed based on comment, thank you.

**Pull request merged**